### PR TITLE
[SPARK-32393][SQL][TEST] Add tests for all the character types in PostgresIntegrationSuite

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This PR intends to add tests to check if all the character types in PostgreSQL supported. 

The document for character types in PostgreSQL: https://www.postgresql.org/docs/current/datatype-character.html

Closes #29192.

### Why are the changes needed?

For better test coverage.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Add tests.

### CODE REVIEW

**Comment by maropu** [comment]
@dongjoon-hyun I checked manually that we've already supported the character types. So, I opened this PR to just add some trivial tests in the integration test. Could you check this?

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/PostgresIntegrationSuite.scala [hunk 1]
```diff
@@ -76,8 +76,13 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationSuite {
       "'172.16.0.42']::inet[], ARRAY['192.168.0.0/24', '10.1.0.0/16']::cidr[], " +
       """ARRAY['{"a": "foo", "b": "bar"}', '{"a": 1, "b": 2}']::json[], """ +
       """ARRAY['{"a": 1, "b": 2, "c": 3}']::jsonb[])"""
-    )
-      .executeUpdate()
+    ).executeUpdate()
```
Comment (maropu) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/PostgresIntegrationSuite.scala:79]: nit: Updated this line to fix the wired format.

**Comment by dongjoon-hyun** [comment]
Oh, thanks, @maropu .

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

Since this is an integration test, I verified it manually.
```
PostgresIntegrationSuite:
- Type mapping for various types
- Basic write test
- Creating a table with shorts and floats
- SPARK-20557: column type TIMESTAMP with TIME ZONE and TIME with TIME ZONE should be recognized
- SPARK-22291: Conversion error when transforming array types of uuid, inet and cidr to StingType in PostgreSQL
- query JDBC option
- write byte as smallint
- character type tests
```

In addition to that, since the IT has a read test only, I verified that Apache Spark 3.0.0 can write to `bpchar` column successfully.

Thank you, @maropu .

Merged to master/3.0.

**Comment by dongjoon-hyun** [comment]
Ur, BTW, while updating JIRA, I found that the original JIRA example is different.
We verified that `c4 bpchar`. But, the original example is `ARRAY`.

**Comment by maropu** [comment]
> Ur, BTW, while updating JIRA, I found that the original JIRA example is different.
We verified that c4 bpchar. But, the original example is ARRAY.

Ur, I see. I misunderstood it a bit.

**Pull request closed without merging**